keps/sig-cli/0014-krew.md

```diff
@@ -0,0 +1,237 @@
+---
+kep-number: 14
+title: Krew
+authors:
+  - "@lbb"
+owning-sig: sig-cli
+participating-sigs:
+  - sig-cli
+reviewers:
+  - TBD
+  - "@pwittroc"
+approvers:
+  - "@pwittroc"
+editor: lbb
+creation-date: 2018-07-03
+last-updated: yyyy-mm-dd
+status: provisional
+see-also:
+  - n/a
+replaces:
+  - n/a
+superseded-by:
+  - n/a
+---
+
+# Krew
+
+## Table of Contents
+* [Krew](#krew)
+  * [Table of Contents](#table-of-contents)
+  * [Summary](#summary)
+  * [Motivation](#motivation)
+    * [Discoverability](#discoverability)
+    * [Cross-platform Packaging](#cross-platform-packaging)
+    * [Lifecycle management](#lifecycle-management)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+  * [Proposal](#proposal)
+      * [Capabilities](#capabilities)
+    * [Scenarios (standalone only)](#scenarios-standalone-only)
+    * [Scenarios](#scenarios)
+    * [UX and CLI Flow](#ux-and-cli-flow)
+    * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints-optional)
+    * [Risks](#risks)
+  * [Graduation Criteria](#graduation-criteria)
+  * [Implementation History](#implementation-history)
+  * [Drawbacks](#drawbacks)
+
+## Summary
+
+krew is a package manager for the kubectl plugin system. krew introduces a way
+to handle kubectl plugin management and discovery.
+
+## Motivation
+
+The kubectl plugin system allows users to extend kubectl by placing extension
+executables in a plugin directory. This allows new subcommands and subcommand
+trees to be added in the kubectl CLI interface under `kubectl plugin <NAME>`.
```

Comment:
let's really see if we can remove the "plugin" word from the cmd line - that's a pretty big UX issue

Reply:
save the "plugin" word for managing plugins

Reply:
That makes sense. I think that issue is out of scope for this KEP and is related to https://github.com/kubernetes/community/blob/a6dcf8678b5c3febaacf079b28174b4a2cbd88e2/contributors/design-proposals/cli/kubectl-extension.md